spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileWritingMessageHandlerSpec.java

```diff
@@ -0,0 +1,191 @@
+import org.springframework.integration.file.support.FileExistsMode;
+import org.springframework.messaging.Message;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.util.Assert;
+
+/**
+ * @author Artem Bilan
+ *
+ * @since 5.0
+ */
+public class FileWritingMessageHandlerSpec
+		extends MessageHandlerSpec<FileWritingMessageHandlerSpec, FileWritingMessageHandler>
+		implements ComponentsRegistration {
+
+	private FileNameGenerator fileNameGenerator;
+
+	private DefaultFileNameGenerator defaultFileNameGenerator;
+
+	FileWritingMessageHandlerSpec(File destinationDirectory) {
+		this.target = new FileWritingMessageHandler(destinationDirectory);
+	}
+
+	FileWritingMessageHandlerSpec(String directoryExpression) {
+		this(PARSER.parseExpression(directoryExpression));
+	}
+
+	<P> FileWritingMessageHandlerSpec(Function<Message<P>, ?> directoryFunction) {
+		this(new FunctionExpression<>(directoryFunction));
+	}
+
+	FileWritingMessageHandlerSpec(Expression directoryExpression) {
+		this.target = new FileWritingMessageHandler(directoryExpression);
+	}
+
+	FileWritingMessageHandlerSpec expectReply(boolean expectReply) {
+		this.target.setExpectReply(expectReply);
+		if (expectReply) {
+			this.target.setRequiresReply(true);
+		}
+		return _this();
+	}
+
+	public FileWritingMessageHandlerSpec autoCreateDirectory(boolean autoCreateDirectory) {
+		this.target.setAutoCreateDirectory(autoCreateDirectory);
+		return _this();
+	}
+
+	public FileWritingMessageHandlerSpec temporaryFileSuffix(String temporaryFileSuffix) {
+		this.target.setTemporaryFileSuffix(temporaryFileSuffix);
+		return _this();
+	}
+
+	public FileWritingMessageHandlerSpec fileExistsMode(FileExistsMode fileExistsMode) {
+		this.target.setFileExistsMode(fileExistsMode);
+		return _this();
+	}
+
+	public FileWritingMessageHandlerSpec fileNameGenerator(FileNameGenerator fileNameGenerator) {
+		this.fileNameGenerator = fileNameGenerator;
+		this.target.setFileNameGenerator(fileNameGenerator);
+		return _this();
+	}
+
+	public FileWritingMessageHandlerSpec fileNameExpression(String fileNameExpression) {
+		Assert.isNull(this.fileNameGenerator,
+				"'fileNameGenerator' and 'fileNameGeneratorExpression' are mutually exclusive.");
+		this.defaultFileNameGenerator = new DefaultFileNameGenerator();
+		this.defaultFileNameGenerator.setExpression(fileNameExpression);
+		return fileNameGenerator(this.defaultFileNameGenerator);
+	}
+
+
+	public FileWritingMessageHandlerSpec deleteSourceFiles(boolean deleteSourceFiles) {
+		this.target.setDeleteSourceFiles(deleteSourceFiles);
+		return _this();
+	}
+
+	public FileWritingMessageHandlerSpec charset(String charset) {
+		this.target.setCharset(charset);
+		return _this();
+	}
+
+	/**
+	 * @param appendNewLine true if a new-line should be written to the file after payload is written.
+	 * @return the spec.
+	 * @see FileWritingMessageHandler#setAppendNewLine(boolean)
+	 */
```

Comment:
We're a bit inconsistent here with some methods having Javadoc, others not (and none in the previous spec).

I am not saying we have to address it now, but we should make sure when we do docs for RC/GA that we addres these too.